app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -524,6 +529,34 @@ public Cursor runQuery(CharSequence charSequence) {
                 handleSentImage(intent);
             }
         }
+
+        handleOnFragmentChange();
+        calendar = Calendar.getInstance();
+    }
+
+    private void handleOnFragmentChange() {
+        getSupportFragmentManager().addOnBackStackChangedListener(new FragmentManager.OnBackStackChangedListener() {
+            @Override
+            public void onBackStackChanged() {
+                FragmentManager fragmentManager = getSupportFragmentManager();
+                Fragment fragment = fragmentManager.findFragmentById(R.id.list_fragment_container);
+                if (fragment instanceof MessageListFragment)
+                {
+                    String tag = fragment.getTag();
+                    if (menu==null)
+                        return;
+                    if (tag.equals(getString(R.string.home_frament_tag)))
+                    {
+                        calendar = Calendar.getInstance();
+                        menu.getItem(2).getSubMenu().getItem(0).setTitle(R.string.menu_today);
+                    }else if (tag.equals(getString(R.string.today_fragment_tag)))
```

Comment:
Rather than saving TAGS for fragments in the Strings.xml it would be better to create file in the utils folder for constants.